File: src/main/java/it/esinware/biblioteca/model/BookModel.java
package it.esinware.biblioteca.model;

import it.esinware.biblioteca.domain.Book;
import it.esinware.mapping.annotation.TypeBinding;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@TypeBinding(binding = Book.class)
public class BookModel {
	
	private String isbn;
	private String title;
	private Long price;
	private Boolean isAvailable;	
	private EditorModel editor;	
	private AuthorModel author;
	private ClientModel client;

}

File: ui/src/app/model/book.model.ts
import { Type } from 'class-transformer';
import 'reflect-metadata';

import { Author } from '../model/author.model';
import { Editor } from '../model/editor.model';
import { Client } from '../model/client.model';

export class Book {
	
	public isbn : string;
	public title: string;
	public price: number;
	public isAvailable: boolean;
	@Type(() => Editor)	
	public editor: Editor;
	@Type(() => Author)
	public author: Author;
	@Type(() => Client)
	public client: Client;
	
}